src/github/github-event.service.ts

```diff
@@ -35,25 +35,25 @@ export class GithubEventService {
 
     if (populate) {
       const createdEvent = await newEvent.save();
-      return await createdEvent.populate('githubUsername', '-__v');
+      return createdEvent.populate('githubUsername', '-__v');
     }
 
-    return await newEvent.save();
+    return newEvent.save();
```

Comment:
I think the `await` is important otherwise the API will return a `200` but the data might not have been saved yet

Reply:
Thanks for your feedback. I've made an update. Please have a look when you have a moment

Reply:
Thanks James! But I think the same applies to all the `await`s that were removed, I think they are all still needed

Reply:
Btw I was just reading the link you shared in issue #292 , I would be interested to get @Cahllagerfeld thoughts on it

Reply:
Sure, the thing with `return await` is that it is redundant outside of a try/catch block and is considered by some people a bad practice. You have tests which should alieviate your concerns. 

Of course it doesn't necessarily "break" your code, it does have a slight performance impact, that said, I'm also happy for this PR to be closed also 👍  

But I'll wait to see @Cahllagerfeld 's thoughts

Reply:
Thanks James for explaining 👍 . I was not aware of this performance issue, so I need to do more research to understand better.

Plus I feel you might have highlighted something else, maybe we should be wrapping these in `try/catch` 🤔  (whatever happens, I feel I have/will learn something new, so thank you 🔥 )

Reply:
I never heard of this before either, so I can't give a statement on this... The errors are handled inside the Services, so there won't be anything to catch on Controller-Level (at least I hope so). I didn't check this issue in detail yet, would also need to do some research on it.